File: lib/screens/items/create_item_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:bmstally_app/models/item.dart';
import 'package:bmstally_app/providers/items_provider.dart';

class CreateItemScreen extends StatefulWidget {
  const CreateItemScreen({super.key});

  @override
  State<CreateItemScreen> createState() => _CreateItemScreenState();
}

class _CreateItemScreenState extends State<CreateItemScreen> {
  final _formKey = GlobalKey<FormState>();
  final _nameController = TextEditingController();
  final _descController = TextEditingController();
  final _openingStockController = TextEditingController();
  final _purchaseRateController = TextEditingController();
  final _salesRateController = TextEditingController();

  String? _category;
  String? _group;
  String? _unit;

  @override
  void dispose() {
    _nameController.dispose();
    _descController.dispose();
    _openingStockController.dispose();
    _purchaseRateController.dispose();
    _salesRateController.dispose();
    super.dispose();
  }

  void _save() {
    if (!_formKey.currentState!.validate()) return;
    if (_category == null || _unit == null) {
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(content: Text('Please fill all required fields')),
      );
      return;
    }

    final provider = context.read<ItemsProvider>();
    final item = Item(
      id: DateTime.now().millisecondsSinceEpoch.toString(),
      name: _nameController.text.trim(),
      category: _category!,
      group: _group ?? '',
      unit: _unit!,
      openingStock: double.tryParse(_openingStockController.text) ?? 0,
      quantity: double.tryParse(_openingStockController.text) ?? 0,
      purchaseRate: double.tryParse(_purchaseRateController.text) ?? 0,
      salesRate: double.tryParse(_salesRateController.text) ?? 0,
      description: _descController.text.trim(),
    );
    provider.addItem(item);
    Navigator.of(context).pop();
    ScaffoldMessenger.of(context).showSnackBar(
      const SnackBar(content: Text('Item saved successfully')),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Colors.blue[700],
        foregroundColor: Colors.white,
        title: const Text('Create Item'),
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(24),
        child: Form(
          key: _formKey,
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              const Text('Item Information',
                  style: TextStyle(fontWeight: FontWeight.w600, fontSize: 16)),
              const SizedBox(height: 16),
              TextFormField(
                controller: _nameController,
                decoration: const InputDecoration(
                  labelText: 'Item Name *',
                  prefixIcon: Icon(Icons.inventory_2),
                  border: OutlineInputBorder(),
                ),
                validator: (v) => v == null || v.trim().isEmpty ? 'Required' : null,
              ),
              const SizedBox(height: 16),
              DropdownButtonFormField<String>(
                initialValue: _category,
                decoration: const InputDecoration(
                  labelText: 'Category *',
                  prefixIcon: Icon(Icons.category),
                  border: OutlineInputBorder(),
                ),
                items: ['Building Material', 'Paint', 'Plumbing', 'Tiles', 'Electrical', 'Sanitary']
                    .map((c) => DropdownMenuItem(value: c, child: Text(c)))
                    .toList(),
                onChanged: (v) => setState(() => _category = v),
              ),
              const SizedBox(height: 16),
              DropdownButtonFormField<String>(
                initialValue: _group,
                decoration: const InputDecoration(
                  labelText: 'Group',
                  prefixIcon: Icon(Icons.group_work),
                  border: OutlineInputBorder(),
                ),
                items: ['Raw Material', 'Finished Good', 'Trading Good', 'Consumable']
                    .map((g) => DropdownMenuItem(value: g, child: Text(g)))
                    .toList(),
                onChanged: (v) => setState(() => _group = v),
              ),
              const SizedBox(height: 16),
              DropdownButtonFormField<String>(
                initialValue: _unit,
                decoration: const InputDecoration(
                  labelText: 'Unit *',
                  prefixIcon: Icon(Icons.scale),
                  border: OutlineInputBorder(),
                ),
                items: ['Nos', 'Kg', 'Pcs', 'Bags', 'Box', 'Roll', 'Ton', 'Set', 'Litre']
                    .map((u) => DropdownMenuItem(value: u, child: Text(u)))
                    .toList(),
                onChanged: (v) => setState(() => _unit = v),
                validator: (v) => v == null ? 'Required' : null,
              ),
              const SizedBox(height: 16),
              TextFormField(
                controller: _openingStockController,
                keyboardType: TextInputType.number,
                decoration: const InputDecoration(
                  labelText: 'Opening Stock',
                  prefixIcon: Icon(Icons.storage),
                  border: OutlineInputBorder(),
                ),
              ),
              const SizedBox(height: 16),
              TextFormField(
                controller: _purchaseRateController,
                keyboardType: TextInputType.number,
                decoration: const InputDecoration(
                  labelText: 'Purchase Rate',
                  prefixIcon: Icon(Icons.shopping_cart),
                  border: OutlineInputBorder(),
                ),
              ),
              const SizedBox(height: 16),
              TextFormField(
                controller: _salesRateController,
                keyboardType: TextInputType.number,
                decoration: const InputDecoration(
                  labelText: 'Sales Rate',
                  prefixIcon: Icon(Icons.attach_money),
                  border: OutlineInputBorder(),
                ),
              ),
              const SizedBox(height: 16),
              TextFormField(
                controller: _descController,
                maxLines: 3,
                decoration: const InputDecoration(
                  labelText: 'Description',
                  prefixIcon: Icon(Icons.description),
                  border: OutlineInputBorder(),
                ),
              ),
              const SizedBox(height: 32),
              SizedBox(
                width: double.infinity,
                height: 48,
                child: FilledButton(
                  onPressed: _save,
                  child: const Text('Save Item'),
                ),
              ),
              const SizedBox(height: 12),
              SizedBox(
                width: double.infinity,
                height: 48,
                child: OutlinedButton(
                  onPressed: () => Navigator.of(context).pop(),
                  child: const Text('Cancel'),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

File: app/src/main/java/com/bmstally/app/ui/screens/CreateItemScreen.kt
package com.bmstally.app.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.bmstally.app.data.MockDataService
import com.bmstally.app.model.Item
import com.bmstally.app.viewmodel.ItemsViewModel
import java.util.UUID

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun CreateItemScreen(
    tenantId: String,
    viewModel: ItemsViewModel,
    onBack: () -> Unit
) {
    var name by remember { mutableStateOf("") }
    var description by remember { mutableStateOf("") }
    var openingStock by remember { mutableStateOf("") }
    var purchaseRate by remember { mutableStateOf("") }
    var salesRate by remember { mutableStateOf("") }
    var selectedCategory by remember { mutableStateOf<String?>(null) }
    var selectedGroup by remember { mutableStateOf<String?>(null) }
    var selectedUnit by remember { mutableStateOf<String?>(null) }
    var categoryExpanded by remember { mutableStateOf(false) }
    var groupExpanded by remember { mutableStateOf(false) }
    var unitExpanded by remember { mutableStateOf(false) }
    var errorMsg by remember { mutableStateOf<String?>(null) }

    val categories = listOf("Building Material", "Paint", "Plumbing", "Tiles", "Electrical", "Sanitary")
    val groups = listOf("Raw Material", "Finished Good", "Trading Good", "Consumable")
    val units = MockDataService.units

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Create Item", color = Color.White) },
                navigationIcon = { IconButton(onBack) { Icon(Icons.Default.ArrowBack, "Back", tint = Color.White) } },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = Color(0xFF3F51B5))
            )
        }
    ) { padding ->
        Column(
            Modifier.fillMaxSize().padding(padding).verticalScroll(rememberScrollState()).padding(24.dp)
        ) {
            Text("Item Information", fontWeight = FontWeight.W600, fontSize = 16.sp)
            Spacer(Modifier.height(16.dp))

            OutlinedTextField(
                value = name,
                onValueChange = { name = it; errorMsg = null },
                label = { Text("Item Name *") },
                leadingIcon = { Icon(Icons.Default.Inventory2, null) },
                singleLine = true,
                modifier = Modifier.fillMaxWidth()
            )

            Spacer(Modifier.height(16.dp))

            ExposedDropdownMenuBox(categoryExpanded, { categoryExpanded = it }) {
                OutlinedTextField(
                    value = selectedCategory ?: "",
                    onValueChange = {},
                    readOnly = true,
                    label = { Text("Category *") },
                    leadingIcon = { Icon(Icons.Default.Category, null) },
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(categoryExpanded) },
                    modifier = Modifier.fillMaxWidth().menuAnchor(),
                    singleLine = true
                )
                ExposedDropdownMenu(categoryExpanded, { categoryExpanded = false }) {
                    categories.forEach { c ->
                        DropdownMenuItem(
                            text = { Text(c) },
                            onClick = { selectedCategory = c; categoryExpanded = false }
                        )
                    }
                }
            }

            Spacer(Modifier.height(16.dp))

            ExposedDropdownMenuBox(groupExpanded, { groupExpanded = it }) {
                OutlinedTextField(
                    value = selectedGroup ?: "",
                    onValueChange = {},
                    readOnly = true,
                    label = { Text("Group") },
                    leadingIcon = { Icon(Icons.Default.GroupWork, null) },
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(groupExpanded) },
                    modifier = Modifier.fillMaxWidth().menuAnchor(),
                    singleLine = true
                )
                ExposedDropdownMenu(groupExpanded, { groupExpanded = false }) {
                    groups.forEach { g ->
                        DropdownMenuItem(
                            text = { Text(g) },
                            onClick = { selectedGroup = g; groupExpanded = false }
                        )
                    }
                }
            }

            Spacer(Modifier.height(16.dp))

            ExposedDropdownMenuBox(unitExpanded, { unitExpanded = it }) {
                OutlinedTextField(
                    value = selectedUnit ?: "",
                    onValueChange = {},
                    readOnly = true,
                    label = { Text("Unit *") },
                    leadingIcon = { Icon(Icons.Default.Scale, null) },
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(unitExpanded) },
                    modifier = Modifier.fillMaxWidth().menuAnchor(),
                    singleLine = true
                )
                ExposedDropdownMenu(unitExpanded, { unitExpanded = false }) {
                    units.forEach { u ->
                        DropdownMenuItem(
                            text = { Text(u) },
                            onClick = { selectedUnit = u; unitExpanded = false }
                        )
                    }
                }
            }

            Spacer(Modifier.height(16.dp))

            OutlinedTextField(
                value = openingStock,
                onValueChange = { openingStock = it },
                label = { Text("Opening Stock") },
                leadingIcon = { Icon(Icons.Default.Storage, null) },
                singleLine = true,
                modifier = Modifier.fillMaxWidth()
            )

            Spacer(Modifier.height(16.dp))

            OutlinedTextField(
                value = purchaseRate,
                onValueChange = { purchaseRate = it },
                label = { Text("Purchase Rate") },
                leadingIcon = { Icon(Icons.Default.ShoppingCart, null) },
                singleLine = true,
                modifier = Modifier.fillMaxWidth()
            )

            Spacer(Modifier.height(16.dp))

            OutlinedTextField(
                value = salesRate,
                onValueChange = { salesRate = it },
                label = { Text("Sales Rate") },
                leadingIcon = { Icon(Icons.Default.AttachMoney, null) },
                singleLine = true,
                modifier = Modifier.fillMaxWidth()
            )

            Spacer(Modifier.height(16.dp))

            OutlinedTextField(
                value = description,
                onValueChange = { description = it },
                label = { Text("Description") },
                leadingIcon = { Icon(Icons.Default.Description, null) },
                modifier = Modifier.fillMaxWidth().height(100.dp)
            )

            if (errorMsg != null) {
                Spacer(Modifier.height(8.dp))
                Text(errorMsg!!, color = Color(0xFFD32F2F), fontSize = 13.sp)
            }

            Spacer(Modifier.height(32.dp))

            Button(
                onClick = {
                    if (name.isBlank() || selectedCategory == null || selectedUnit == null) {
                        errorMsg = "Please fill all required fields"
                        return@Button
                    }
                    val item = Item(
                        id = UUID.randomUUID().toString(),
                        name = name.trim(),
                        category = selectedCategory!!,
                        group = selectedGroup ?: "",
                        unit = selectedUnit!!,
                        quantity = openingStock.toIntOrNull() ?: 0,
                        openingStock = openingStock.toIntOrNull() ?: 0,
                        purchaseRate = purchaseRate.toIntOrNull() ?: 0,
                        salesRate = salesRate.toIntOrNull() ?: 0,
                        reorderLevel = 0
                    )
                    viewModel.addItem(item)
                    onBack()
                },
                modifier = Modifier.fillMaxWidth().height(48.dp)
            ) {
                Text("Save Item")
            }

            Spacer(Modifier.height(12.dp))

            OutlinedButton(
                onClick = onBack,
                modifier = Modifier.fillMaxWidth().height(48.dp)
            ) {
                Text("Cancel")
            }
        }
    }
}